og_ui/src/Form/OgRoleForm.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Drupal\og_ui\Form;
+
+use Drupal\Core\Entity\EntityForm;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\og\Entity\OgRole;
+use Drupal\og\OgRoleInterface;
+
+/**
+ * Form to add or edit an OG role.
+ */
+class OgRoleForm extends EntityForm {
+
+  /**
+   * The entity being used by this form.
+   *
+   * @var \Drupal\og\Entity\OgRole
+   */
+  protected $entity;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $og_role = $this->entity;
+    $form['label'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Role name'),
+      '#default_value' => $og_role->label(),
+      '#size' => 30,
+      '#required' => TRUE,
+      '#maxlength' => 64,
+      '#description' => $this->t('The name for this role. Example: "Moderator", "Editorial board", "Site architect".'),
+    ];
+
+    $role_id = '';
+    if ($og_role->id()) {
+      // The actual role id is <ENTITY TYPE>-<BUNDLE>-<ID>
+      // Given the machine_name constraints, we go back to ID here
+      // as the full id is assembled in OgRole::save().
+      list(, , $role_id) = explode('-', $og_role->id(), 3);
+    }
```

Comment:
The last part of the ID is called the "name" and can be retrieved by calling `$og_role->getName()`. This is more reliable than exploding on the hyphens.

I would also rename the variable from `$role_id` to `$role_name` so that it is clear this contains the name and not the ID.